borgmatic/config/validate.py

```diff
@@ -138,9 +138,15 @@ def normalize_repository_path(repository):
 
 def repositories_match(first, second):
     '''
-    Given two repository paths (relative and/or absolute), return whether they match.
-    '''
-    return normalize_repository_path(first) == normalize_repository_path(second)
+    Given two repository dicts with keys 'path' (relative and/or absolute),
+    and 'label', return whether they match.
+    '''
+    if isinstance(first, str) and isinstance(second, str):
+        return normalize_repository_path(first) == normalize_repository_path(second)
+    elif isinstance(first, dict) and isinstance(second, str):
+        return (second == first.get('label')) or (
+            normalize_repository_path(second) == normalize_repository_path(first.get('path'))
+        )
```

Comment:
If there are implicit expectations here about whether the second repository can be a dict or not, I'd recommend changing the function parameter names and docstring accordingly. (E.g., change `first` and `second` to `config_repository` and `repository_name` or something.) Because if I'm reading this right, if you pass in two dicts right now, this function won't work! An alternative would be to handle any `first` and `second` parameters regardless of whether they are dicts or strings.